Fix i18n import path

hexart opened the pull request:
I'm terribly sorry about these mistakes.
That is my first PR, and I‘ve learned a lot from it.
<img width="514" alt="截屏2025-01-20 16 44 42" src="https://github.com/user-attachments/assets/990c4688-d1c8-42b3-a88a-5b26d613cc33" />
<img width="587" alt="截屏2025-01-20 16 44 49" src="https://github.com/user-attachments/assets/a50c88c7-ed0c-4f00-ab41-6cafc883b921" />
<img width="333" alt="截屏2025-01-20 16 44 56" src="https://github.com/user-attachments/assets/819abde6-5421-4187-ad5c-026d554a923f" />
<img width="549" alt="截屏2025-01-20 16 45 01" src="https://github.com/user-attachments/assets/531e089c-c704-4203-b8f4-533f55e19fc7" />
<img width="779" alt="截屏2025-01-20 16 45 13" src="https://github.com/user-attachments/assets/9abfad7e-43d8-43df-8aba-f2bf0926ebf6" />
Now it rendered correctly.

dokterbob commented:
> I'm terribly sorry about these mistakes. That is my first PR, and I‘ve learned a lot from it. <img alt="截屏2025-01-20 16 44 42" width="514" src="https://private-user-images.githubusercontent.com/5528016/404802287-990c4688-d1c8-42b3-a88a-5b26d613cc33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjYwODcsIm5iZiI6MTczNzM2NTc4NywicGF0aCI6Ii81NTI4MDE2LzQwNDgwMjI4Ny05OTBjNDY4OC1kMWM4LTQyYjMtYTg4YS01YjI2ZDYxM2NjMzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDkzNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjVlNGM1NWE0ZTRjNDZjNGFhZDdhNGQ1NzhkODUyMDNmOGUwN2JlY2NhY2VkNWM0ZDJkMDFmNjdmMmRlN2ZkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.rJhXG1MbnzLL-cVmyuM9R_KJfVAB5L8d93Bd9NedNNo"> <img alt="截屏2025-01-20 16 44 49" width="587" src="https://private-user-images.githubusercontent.com/5528016/404802324-a50c88c7-ed0c-4f00-ab41-6cafc883b921.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjYwODcsIm5iZiI6MTczNzM2NTc4NywicGF0aCI6Ii81NTI4MDE2LzQwNDgwMjMyNC1hNTBjODhjNy1lZDBjLTRmMDAtYWI0MS02Y2FmYzg4M2I5MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDkzNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDM0OTQwNGEzNzM5ZjhlYzE1YmZkNDQ2ZjJiOTljMDMzNmJmMzljMjkxMjNiZmE2NzkwZGQzYWIzNzQ1YjBiYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fANcDDUp7N3G6KbS3moGh7aw8LBr8XvcADtPxniWpbw"> <img alt="截屏2025-01-20 16 44 56" width="333" src="https://private-user-images.githubusercontent.com/5528016/404802348-819abde6-5421-4187-ad5c-026d554a923f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjYwODcsIm5iZiI6MTczNzM2NTc4NywicGF0aCI6Ii81NTI4MDE2LzQwNDgwMjM0OC04MTlhYmRlNi01NDIxLTQxODctYWQ1Yy0wMjZkNTU0YTkyM2YucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDkzNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjQ2YTBmNTg1NGUxZTVkM2Q2ODEzY2JlZWM4MTA0YjhjY2MyMTVmMjlkM2U2NGU3MzNhYjc5NWU0ZTE0Yzg4ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.QYUCPyBxsDv2lkxaBtKDJSQrSBT2wlTEtX-QhgBtERE"> <img alt="截屏2025-01-20 16 45 01" width="549" src="https://private-user-images.githubusercontent.com/5528016/404802415-531e089c-c704-4203-b8f4-533f55e19fc7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjYwODcsIm5iZiI6MTczNzM2NTc4NywicGF0aCI6Ii81NTI4MDE2LzQwNDgwMjQxNS01MzFlMDg5Yy1jNzA0LTQyMDMtYjhmNC01MzNmNTVlMTlmYzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDkzNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDE1NjhmNGU4NTMyNzU0NmU4MGY3OGZhMDdkMTU0MDQ1ZmExZTNjMmFiZjA0ZWZhZTI3ZmVlNDNhYmEyMGUzMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.71QFMcABOawe6zO-_N9DbrnrhDStzAYz_Cd14FEIio8"> <img alt="截屏2025-01-20 16 45 13" width="779" src="https://private-user-images.githubusercontent.com/5528016/404802447-9abfad7e-43d8-43df-8aba-f2bf0926ebf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjYwODgsIm5iZiI6MTczNzM2NTc4OCwicGF0aCI6Ii81NTI4MDE2LzQwNDgwMjQ0Ny05YWJmYWQ3ZS00M2Q4LTQzZGYtOGFiYS1mMmJmMDkyNmViZjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDkzNjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWE1MzdlNDAzYWJkYTZjMDc5Y2ZhOWIyZDVkYzBiZGI2ODNiNDlkMTg0NGIwYmM0ZWM0ZjEyN2Q0ZWE4OTZlYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.CSo-unioTHQ0phOOCZ4oVZw1fGn3IGYAIbgkSQbEpkE">

I can't read Chinese, what does this fix? What's going on here?

dokterbob commented:
(Also, on an internal note, how has our linter not picket this up!??)

hexart commented:
This PR has two main improvements:

1. Fixed inconsistent i18n import methods across files. I found we were using three different ways to import the Translator:
- import { Translator } from '@/components/i18n'
- import { Translator } from '../i18n'
- import Translator from '@/components/i18n/Translator'

After checking through all the files in the project, I found it's better to use relative paths (like '../i18n') for consistency, as most files were already using this pattern.

2. Added internationalization support:
- Added "Rename" and "Delete" strings for the thread options menu
- Added translations for all recently added/modified strings to all supported languages in the project (bn, gu, hi, kn, ml, mr, nl-NL, ta, and te)

(About the linter: Maybe we could add ESLint rules to enforce consistent import paths?)

willydouhard commented:
I am going to open another PR to simplify those paths. They don't make sense anymore since the UI rewrite, closing this one and opening the new one today!

hexart commented:
> I am going to open another PR to simplify those paths. They don't make sense anymore since the UI rewrite, closing this one and opening the new one today!

Cool! Looking forward to it. :)
